diff -up cups-2.4rc1/config-scripts/cups-tls.m4.5~ cups-2.4rc1/config-scripts/cups-tls.m4
--- cups-2.4rc1/config-scripts/cups-tls.m4.5~	2021-11-25 20:08:56.668918055 +0100
+++ cups-2.4rc1/config-scripts/cups-tls.m4	2021-11-25 20:09:18.983006016 +0100
@@ -107,5 +107,5 @@ AC_SUBST([IPPALIASES])
 AC_SUBST([TLSFLAGS])
 AC_SUBST([TLSLIBS])
 
-EXPORT_TLSLIBS="$TLSLIBS"
+EXPORT_TLSLIBS=""
 AC_SUBST([EXPORT_TLSLIBS])
