diff -up cups-2.2.7/scheduler/main.c.1024~ cups-2.2.7/scheduler/main.c
--- cups-2.2.7/scheduler/main.c.1024~	2018-04-30 00:46:26.754047189 +0200
+++ cups-2.2.7/scheduler/main.c	2018-04-30 00:46:26.765046928 +0200
@@ -690,9 +690,15 @@ main(int  argc,				/* I - Number of comm
   */
 
 #ifdef HAVE_ONDEMAND
-  if (OnDemand)
+  if (OnDemand) {
     cupsdAddEvent(CUPSD_EVENT_SERVER_STARTED, NULL, NULL, "Scheduler started on demand.");
-  else
+# ifdef HAVE_SYSTEMD
+    sd_notifyf(0, "READY=1\n"
+		    "STATUS=Scheduler is running...\n"
+		    "MAINPID=%lu",
+		    (unsigned long) getpid());
+# endif /* HAVE_SYSTEMD */
+  } else
 #endif /* HAVE_ONDEMAND */
   if (fg)
     cupsdAddEvent(CUPSD_EVENT_SERVER_STARTED, NULL, NULL, "Scheduler started in foreground.");
diff -up cups-2.2.7/scheduler/org.cups.cupsd.path.in.1024~ cups-2.2.7/scheduler/org.cups.cupsd.path.in
--- cups-2.2.7/scheduler/org.cups.cupsd.path.in.1024~	2018-03-23 04:48:36.000000000 +0100
+++ cups-2.2.7/scheduler/org.cups.cupsd.path.in	2018-04-30 00:46:26.765046928 +0200
@@ -3,7 +3,7 @@ Description=CUPS Scheduler
 PartOf=org.cups.cupsd.service
 
 [Path]
-PathExists=@CUPS_CACHEDIR@/org.cups.cupsd
+PathExistsGlob=@CUPS_REQUESTS@/d*
 
 [Install]
 WantedBy=multi-user.target
diff -up cups-2.2.7/scheduler/org.cups.cupsd.service.in.1024~ cups-2.2.7/scheduler/org.cups.cupsd.service.in
--- cups-2.2.7/scheduler/org.cups.cupsd.service.in.1024~	2018-04-30 00:46:26.765046928 +0200
+++ cups-2.2.7/scheduler/org.cups.cupsd.service.in	2018-04-30 00:47:26.811622413 +0200
@@ -1,10 +1,11 @@
 [Unit]
 Description=CUPS Scheduler
 Documentation=man:cupsd(8)
+After=network.target
 
 [Service]
 ExecStart=@sbindir@/cupsd -l
-Type=simple
+Type=notify
 Restart=always
 
 [Install]
