--- cups-2.2.5/scheduler/main.c.1024~	2017-10-14 13:01:52.418760893 +0200
+++ cups-2.2.5/scheduler/main.c	2017-10-14 13:03:50.500371211 +0200
@@ -690,9 +690,15 @@ main(int  argc,				/* I - Number of comm
   */
 
 #ifdef HAVE_ONDEMAND
-  if (OnDemand)
+  if (OnDemand) {
     cupsdAddEvent(CUPSD_EVENT_SERVER_STARTED, NULL, NULL, "Scheduler started on demand.");
-  else
+# ifdef HAVE_SYSTEMD
+    sd_notifyf(0, "READY=1\n"
+		    "STATUS=Scheduler is running...\n"
+		    "MAINPID=%lu",
+		    (unsigned long) getpid());
+# endif /* HAVE_SYSTEMD */
+  } else
 #endif /* HAVE_ONDEMAND */
   if (fg)
     cupsdAddEvent(CUPSD_EVENT_SERVER_STARTED, NULL, NULL, "Scheduler started in foreground.");
--- cups-2.2.5/scheduler/org.cups.cupsd.path.in.1024~	2017-10-13 20:22:26.000000000 +0200
+++ cups-2.2.5/scheduler/org.cups.cupsd.path.in	2017-10-14 13:01:52.418760893 +0200
@@ -3,7 +3,7 @@ Description=CUPS Scheduler
 PartOf=org.cups.cupsd.service
 
 [Path]
-PathExists=@CUPS_CACHEDIR@/org.cups.cupsd
+PathExistsGlob=@CUPS_REQUESTS@/d*
 
 [Install]
 WantedBy=multi-user.target
--- cups-2.2.5/scheduler/org.cups.cupsd.service.in.1024~	2017-10-13 20:22:26.000000000 +0200
+++ cups-2.2.5/scheduler/org.cups.cupsd.service.in	2017-10-14 13:01:52.418760893 +0200
@@ -1,10 +1,11 @@
 [Unit]
 Description=CUPS Scheduler
 Documentation=man:cupsd(8)
+After=network.target
 
 [Service]
 ExecStart=@sbindir@/cupsd -l
-Type=simple
+Type=notify
 
 [Install]
 Also=org.cups.cupsd.socket org.cups.cupsd.path
