--- cups-1.4svn-r8703/scheduler/cups-driverd.cxx~	2009-06-09 20:43:47.000000000 +0200
+++ cups-1.4svn-r8703/scheduler/cups-driverd.cxx	2009-06-09 21:09:32.000000000 +0200
@@ -211,7 +211,6 @@
 	const char *scheme)		/* I - PPD scheme */
 {
   ppd_info_t	*ppd;			/* PPD */
-  char		*recommended;		/* Foomatic driver string */
 
 
  /*
@@ -250,15 +249,6 @@
   strlcpy(ppd->record.scheme, scheme, sizeof(ppd->record.scheme));
 
  /*
-  * Strip confusing (and often wrong) "recommended" suffix added by
-  * Foomatic drivers...
-  */
-
-  if ((recommended = strstr(ppd->record.make_and_model,
-                            " (recommended)")) != NULL)
-    *recommended = '\0';
-
- /*
   * Add the PPD to the PPD arrays...
   */
 
