diff -up cups-2.2.9/cups/tempfile.c.1019~ cups-2.2.9/cups/tempfile.c
--- cups-2.2.9/cups/tempfile.c.1019~	2018-11-08 21:46:26.951742110 +0100
+++ cups-2.2.9/cups/tempfile.c	2018-11-08 21:50:23.967013083 +0100
@@ -25,6 +25,7 @@
 #  include <io.h>
 #else
 #  include <unistd.h>
+#  include <sys/types.h>
 #endif /* _WIN32 || __EMX__ */
 
 
@@ -45,12 +46,7 @@ cupsTempFd(char *filename,		/* I - Point
 #if defined(__APPLE__) || defined(_WIN32)
   char		tmppath[1024];		/* Temporary directory */
 #endif /* __APPLE__ || _WIN32 */
-#ifdef _WIN32
-  DWORD		curtime;		/* Current time */
-#else
-  struct timeval curtime;		/* Current time */
-#endif /* _WIN32 */
-
+  mode_t	old_umask;		/* Old umask before using mkstemp() */
 
  /*
   * See if TMPDIR is defined...
@@ -114,32 +110,24 @@ cupsTempFd(char *filename,		/* I - Point
     */
 
     snprintf(filename, (size_t)len - 1, "%s/%05lx%08lx", tmpdir, GetCurrentProcessId(), curtime);
-#else
-   /*
-    * Get the current time of day...
-    */
-
-    gettimeofday(&curtime, NULL);
-
-   /*
-    * Format a string using the hex time values...
-    */
-
-    snprintf(filename, (size_t)len - 1, "%s/%05x%08x", tmpdir, (unsigned)getpid(), (unsigned)(curtime.tv_sec + curtime.tv_usec + tries));
-#endif /* _WIN32 */
 
    /*
     * Open the file in "exclusive" mode, making sure that we don't
     * stomp on an existing file or someone's symlink crack...
     */
 
-#ifdef _WIN32
     fd = open(filename, _O_CREAT | _O_RDWR | _O_TRUNC | _O_BINARY,
               _S_IREAD | _S_IWRITE);
-#elif defined(O_NOFOLLOW)
-    fd = open(filename, O_RDWR | O_CREAT | O_EXCL | O_NOFOLLOW, 0600);
 #else
-    fd = open(filename, O_RDWR | O_CREAT | O_EXCL, 0600);
+    /*
+     * Use the standard mkstemp() call to make a temporary filename
+     * securely.  -- andrew.wood@jdplc.com
+     */
+    snprintf(filename, len - 1, "%s/cupsXXXXXX", tmpdir);
+
+    old_umask = umask(0077);
+    fd = mkstemp(filename);
+    umask(old_umask);
 #endif /* _WIN32 */
 
     if (fd < 0 && errno != EEXIST)
