--- cups-1.7b1/cups/http-addr.c.1019~	2013-03-20 19:14:10.000000000 +0100
+++ cups-1.7b1/cups/http-addr.c	2013-04-30 10:42:54.956876751 +0200
@@ -319,7 +319,8 @@ httpAddrLookup(
 
     if (error)
     {
-      if (error == EAI_FAIL)
+      if (error == EAI_FAIL || error == EAI_AGAIN || error == EAI_NODATA ||
+          error == EAI_NONAME)
         cg->need_res_init = 1;
 
       return (httpAddrString(addr, name, namelen));
--- cups-1.7b1/cups/http-addrlist.c.1019~	2013-04-30 10:42:54.956876751 +0200
+++ cups-1.7b1/cups/http-addrlist.c	2013-04-30 10:43:52.770153986 +0200
@@ -581,7 +581,7 @@ httpAddrGetList(const char *hostname,	/*
     }
     else
     {
-      if (error == EAI_FAIL)
+      if (error == EAI_FAIL || error == EAI_AGAIN || error == EAI_NODATA || error == EAI_NONAME)
         cg->need_res_init = 1;
 
       _cupsSetError(IPP_STATUS_ERROR_INTERNAL, gai_strerror(error), 0);
